### DISCUSSION TITLE
upgrade  v3.5-beta1 to  v3.5-beta2 error

### DISCUSSION THREAD

**Discussion by juriaan** [netbox-community/netbox#12352] [Help Wanted!]
- OS: Debian 11.6
- Python 3.9.2
- postgresql 15.2

Started with a fresh copy of v3.5-beta1 and added data, now I want to upgrade to v3.5-beta2, the _upgrade.sh_ script fails

`django.core.exceptions.ImproperlyConfigured: Unable to import plugin nextbox_ui_plugin: Module not found. Check that the plugin module has been installed within the correct Python environment.`

Could fix that by adding:

`pip install nextbox_ui_plugin` to _upggrade.sh_

again run  _upgrade.sh_ and when step **Applying database migrations (python3 netbox/manage.py migrate)...** begins, the following error is thrown

`ImportError: cannot import name 'DynamicModelMultipleChoiceField' from 'utilities.forms' (/opt/netbox-3.5-beta2/netbox/utilities/forms/__init__.py)`

Attached is the full output of the  _upgrade.sh_ script . Does anyone have a clue?

[beta2.error_2.txt](https://github.com/netbox-community/netbox/files/11334788/beta2.error_2.txt)

**Comment by abhi1693** [comment]
> Could fix that by adding:
> pip install nextbox_ui_plugin to upggrade.sh

You should not add anything to upgrade.sh. You should instead create `local_requirements.txt` and add your dependencies in there. upgrade.sh will automatically read the file.

> ImportError: cannot import name 'DynamicModelMultipleChoiceField' from 'utilities.forms' (/opt/netbox-3.5-beta2/netbox/utilities/forms/__init__.py)

There were changes in imports between beta1 and beta2 so that may be causing the issue here. You can see that here: https://github.com/netbox-community/netbox/commit/d470848b2932b7ef2dc68469d11bfa1609d40403

The best way to test beta2 would be to do a fresh installs as it is not guaranteed that upgrades between beta version will work.
